### PR TITLE
Continuing the implementation of initial entries support in p4c

### PR DESCRIPTION
**Pull request by jafingerhut**
These changes update the contents of P4Info and table entries files as generated by p4c, to match what has been agreed upon by the P4 API work group in this pull request
https://github.com/p4lang/p4runtime/pull/432

This PR is intended to make a significant step towards resolving issue https://github.com/p4lang/p4c/issues/4016 but will not do so completely, as this PR does not address the parts of that issue suggesting the creation of STF or PTF tests that exercise this new feature.

### CODE REVIEW

**Comment by jafingerhut** [comment]
Note: This PR will probably fail CI tests with compile-time errors until after this PR has been merged in: https://github.com/p4lang/p4runtime/pull/432

and then after that, it will likely continue to fail until the control-plane/p4runtime sub-module of p4c is updated to a commit SHA that includes the changes in the PR linked above.

**Comment by fruffy** [comment]
I believe #4056 may be needed to be able to update P4Runtime.

**Comment by jafingerhut** [comment]
@rst0git Several hours ago there was a commit to the repo https://github.com/p4lang/p4runtime, which this PR depends upon.  It appears that some of these tests use apt to install p4lang-pi and some other precompiled packages, which may need updating in order for these tests to pass.  Do you know if that is true?

**Comment by rst0git** [comment]
@jafingerhut Would you be able to update the P4Runtime commit in `CMakeLists.txt`?
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index b7bc5620a..c6117c182 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -383,7 +383,7 @@ set(FETCHCONTENT_QUIET OFF)
 FetchContent_Declare(
   p4runtime
   GIT_REPOSITORY https://github.com/p4lang/p4runtime.git
-  GIT_TAG        d76a3640a223f47a43dc34e5565b72e43796ba57
+  GIT_TAG        1e771c4e05c4e7e250df00212b3ca02ee3202d71
   GIT_PROGRESS TRUE
 )
 FetchContent_MakeAvailable(p4runtime)
```

**Comment by jafingerhut** [comment]
@rst0git There isn't any big urgency, but if you happen to have some time and interest, do you have any knowledge of why so many of the tests would be failing with the changes in this PR?  I have rebased it to latest main version as of 2023-Aug-15, plus my few changes.

**Comment by jafingerhut** [comment]
@rst0git Sorry for the noise, but I believe I may have discovered the reasons for failing tests myself -- good old fashioned "run all the tests locally on your own system and debug them" approach worked.  Sigh.

File: bazel/p4c_deps.bzl
Comment (rst0git) [inline bazel/p4c_deps.bzl:46]:
```Suggestion
            # Newest commit on main branch as of August 11, 2023.
```

**Review by rst0git — APPROVED** [review]

**Comment by jafingerhut** [comment]
Got one approval from a P4 API work group co-chair in a comment, and another approval from p4c developer, so merging.

**Pull request merged**
